src/region/RegionAnnotations.tsx

```diff
@@ -1,83 +1,130 @@
 import * as React from 'react';
-import { Annotation, Rect, TargetRegion } from '../@types';
+import merge from 'lodash/merge';
 import PopupReply from '../components/Popups/PopupReply';
 import RegionAnnotation from './RegionAnnotation';
 import RegionCreator from './RegionCreator';
+import { Annotation, Rect, NewAnnotation, TargetRegion } from '../@types';
+import { CreatorItem } from '../store/creator';
 import './RegionAnnotations.scss';
 
 type Props = {
     annotations: Annotation[];
     isCreating: boolean;
+    page: number;
+    scale: number;
+    saveAnnotation: (annotation: NewAnnotation) => void;
+    setStaged: (staged: CreatorItem | null) => void;
+    setStatus: (status: string) => void;
+    staged?: CreatorItem | null;
+    status: string | null;
 };
 
 type State = {
-    current: Partial<Rect> | null;
-    hasComment: boolean;
-    hasDrawn: boolean;
+    targetRef?: HTMLAnchorElement;
 };
 
 export default class RegionAnnotations extends React.Component<Props, State> {
     static defaultProps = {
         annotations: [],
         isCreating: false,
+        scale: 1,
     };
 
-    targetRef: React.RefObject<HTMLAnchorElement> = React.createRef();
+    state: State = {};
 
-    state: State = {
-        current: null, // TODO: Move unsaved region state to central store
-        hasComment: false,
-        hasDrawn: false,
-    };
+    setStaged(data: Partial<CreatorItem>): void {
+        const { staged, page, setStaged } = this.props;
+        setStaged(merge({ location: page }, staged, data));
```

Comment:
Should location be the last in `merge` to override whatever may already be in `staged`?

Reply:
Yeah, I'll refine this a bit.